package/harvester-os/files/usr/sbin/harv-update-rke2-server-url

```diff
@@ -6,17 +6,24 @@ RKE2_VIP_CONFIG_FILE=/etc/rancher/rke2/config.yaml.d/90-harvester-vip.yaml
 
 update_agent_conf()
 {
-  export SERVER_URL="$(yq -e e .serverurl /oem/harvester.config | sed -E 's,^https://(.*):8443,https://\1:9345,')"
+  server_url=$(yq -e e .serverurl /oem/harvester.config)
 
-  if [ -z "$SERVER_URL" ]; then
-    echo "[Error] fail to get server URL with VIP."
+  if [ -z "$server_url" ]; then
+    echo "[Error] fail to get server URL in Harvester config."
     exit 1
   fi
 
+  if ! echo $server_url | grep -E '^https://(.*):(8443|443)'; then
```

Comment:
Is it valid to use a server url without the port?

Reply:
If the port is not included, rancherd can't get any needed info to join a cluster. Thus RKE2 will not be deployed and this script won't be run.

There are two places that involve the server URL:
- Text UI path: port is always appended to VIP.
- PXE boot path: the user needs to explicitly specify the full URL including the port number.